examples/03_connectivity/plot_confounds_removal.py

```diff
@@ -0,0 +1,167 @@
+        percentile=100.)
+    confounds = np.hstack((confounds, tissue_confounds))
+
+#######################################################
+# Exploring confounds contribution to GM voxels signals
+# -----------------------------------------------------
+# Typically motion has a dramatic impact on voxels within the contour of the
+# brain.
+
+#######################################################################
+# We compute GM voxels signals
+from nilearn import input_data
+gm_masker = input_data.NiftiMasker(mask_img=mask_images['gm'],
+                                   memory='nilearn_cache',
+                                   standardize=True)
+gm_signals = gm_masker.fit_transform(func_filename)
+
+#######################################################################
+# and predict voxels signals from confounds.
+from sklearn import linear_model
+regr = linear_model.LinearRegression(normalize=True)
+regr.fit(confounds, gm_signals)
+total_variance = np.sum(gm_signals ** 2, axis=0)
+residual_variance = np.sum((regr.predict(confounds) - gm_signals) ** 2,
+                           axis=0)
+variance_percent = 100. - 100. * residual_variance / total_variance
+# Transform from array to image
+variance_img = gm_masker.inverse_transform(variance_percent)
+
+#######################################################################
+# The variance map shows the percent of variance explained by confounds.
+display = plotting.plot_stat_map(variance_img, bg_img=mean_func_img,
+                                 cut_coords=cut_coords, vmax=100.)
+
+############################################################################
+# Visualizing the impact of confounds removal on voxel-to-voxel connectivity
+# --------------------------------------------------------------------------
+# After confounds removal, the distribution of connectivity values should be
+# tight and approximately centered on zero.
+
+#######################################################################
+# We use the same masker, but include confounds.
+cleaned_gm_signals = gm_masker.fit_transform(func_filename,
+                                             confounds=confounds)
+
+#######################################################################
+# We plot the histogram of voxel-to-voxel correlations for a selection of
+# voxels.
+import matplotlib.pylab as plt
+selected_voxels = range(0, gm_signals.shape[1], 10)
+plt.figure(figsize=(8, 4))
+for signals, label, color in zip([gm_signals, cleaned_gm_signals],
+                                 ['with confounds', 'without confounds'],
+                                 'rb'):
+    selected_signals = signals[:, selected_voxels]
+    correlations = np.dot(selected_signals.T, selected_signals) / \
+        selected_signals.shape[0]
+    plt.hist(correlations[np.triu_indices_from(correlations, k=1)],
+             color=color, alpha=.3, bins=100, lw=0, label=label)
+plt.legend()
+plt.xlabel('correlation values')
+plt.show()
```

Comment:
Wouldn't it be more insightful to compute correlation before and after confound removal ?